### PR TITLE
Adding 3 yosensi devices to Akenza platform

### PR DESCRIPTION
**Pull request by maciejSAC**
Hello, I've completed the integration of 3 yosensi devices: YO Temp, YO Pure Pro, YO Refrigerant Monitor.

### CODE REVIEW

File: types/yosensi/YO-PurePro/co.schema.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "title": "CO",
+    "properties": {
+      "co": {
+        "title": "CO Concentration",
+        "type": "number",
+        "description": "CO Concentration in ppm.",
+        "unit": "ppm",
+        "minimum": 0,
+        "maximum": 1000,
+        "hideFromKpis": true
```
Comment (mariusgiger) [inline types/yosensi/YO-PurePro/co.schema.json:16]: setting hideFromKpis to true will disable displaying the value on the device details page in the akenza platform, we therefore recommend to only set this to true if you would like to hide the value (e.g. internal or debugging information)

File: types/yosensi/YO-PurePro/meta.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "Illuminance",
+    "TVOC",
+    "CO",
+    "CO2",
+    "Sound Pressure Level",
+    "PM2.5",
+    "PM4",
+    "PM10"
+  ],
+  "outputTopics": [
```
Comment (mariusgiger) [inline types/yosensi/YO-PurePro/meta.json:22]: Currently a topic is used for each data point. We recommend grouping data points that belong together and are sent in the same LoRa uplink into the same topic; e.g. topic `iaq` containing tvoc, co, co2, or `environment` containing all data keys. We usually distinguish between a `lifecycle` topic containing system information and a measurement topic containing the values of interest.

File: types/yosensi/YO-PurePro/uplink.js [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+    }
+    
+    return {
+        data
+    };
+}
+
+function consume(event) {
+    const payloadData = event.data.payloadHex;
+    const {port} = event.data;
```
Comment (mariusgiger) [inline types/yosensi/YO-PurePro/uplink.js:530]: we recommend using the [prettier Vs Code extension](https://prettier.io/docs/en/editors.html) to align formatting. If you are not using this IDE, you can run `npx prettier ./types/yosensi --write`

**Review by mariusgiger — COMMENTED** [review]
Hi @maciejSAC, thanks a lot for your contribution! This looks great and we are happy to add these types to the akenza library. 

I have added two comments regarding topic structuring (if possible try to group data that is sent in the same uplink into the same topic, which will make data loading more efficient for API users) and formatting (if possible run prettier, otherwise we will do it).

Cheers,
M.

File: types/yosensi/YO-PurePro/uplink.js (hunk 3 above)
Comment (maciejSAC) [reply types/yosensi/YO-PurePro/uplink.js:530]: Thank you for the tip, I will do this right away with the devices I sent and with all the future devices.

File: types/yosensi/YO-PurePro/meta.json (hunk 2 above)
Comment (maciejSAC) [reply types/yosensi/YO-PurePro/meta.json:22]: I've been trying to handle varying payloads from our device, but I'm running into issues with passing the tests. For instance, our payloads can sometimes look like this:
Payload 1: 'co2', 'pm10'
Payload 2: 'co', 'pm4', 'sound pressure', ...
Payload 3: 'co'

I haven't been able to come up with a solution that manages this variability effectively, so I ended up handling it in a different way. If you have any tips or ideas on how to better manage these diverse payloads, I'd really appreciate it.

File: types/yosensi/YO-PurePro/co.schema.json (hunk 1 above)
Comment (maciejSAC) [reply types/yosensi/YO-PurePro/co.schema.json:16]: Alright, thanks for clarification, I will do the changes and push them together with the formatted documents. (using Prettier)

File: types/yosensi/YO-PurePro/meta.json (hunk 2 above)
Comment (mariusgiger) [reply types/yosensi/YO-PurePro/meta.json:22]: ok thanks for the clarification, in this case it makes sense to split the payloads

**Review by mariusgiger — APPROVED** [review]
thanks a lot for the adjustments, I will now proceed to merge these device types and they will become available in akenza shortly

**Pull request merged**